pkgs/development/libraries/onnxruntime/default.nix

```diff
@@ -59,30 +62,36 @@ let
     owner = "pytorch";
     repo = "cpuinfo";
     # There are no tags in the repository
-    rev = "5916273f79a21551890fd3d56fc5375a78d1598d";
-    hash = "sha256-nXBnloVTuB+AVX59VDU/Wc+Dsx94o92YQuHp3jowx2A=";
+    rev = "3c8b1533ac03dd6531ab6e7b9245d488f13a82a5";
+    hash = "sha256-eshoHmGiu5k0XE/A1SWf7OvBj7/YD9JNSZgoyGzGcLA=";
   };
 
-  flatbuffers = fetchFromGitHub {
-    owner = "google";
-    repo = "flatbuffers";
-    rev = "v1.12.0";
-    hash = "sha256-L1B5Y/c897Jg9fGwT2J3+vaXsZ+lfXnskp8Gto1p/Tg=";
+  pytorch_clog = effectiveStdenv.mkDerivation {
+    pname = "clog";
+    version = "3c8b153";
+    src = "${pytorch_cpuinfo}/deps/clog";
+
+    nativeBuildInputs = [ cmake gbenchmark gtest ];
+    cmakeFlags = [
+      "-DUSE_SYSTEM_GOOGLEBENCHMARK=ON"
+      "-DUSE_SYSTEM_GOOGLETEST=ON"
+      "-DUSE_SYSTEM_LIBS=ON"
+    ];
   };
 
   onnx = fetchFromGitHub {
     owner = "onnx";
     repo = "onnx";
-    rev = "refs/tags/v1.14.1";
-    hash = "sha256-ZVSdk6LeAiZpQrrzLxphMbc1b3rNUMpcxcXPP8s/5tE=";
+    rev = "refs/tags/v1.16.1";
+    hash = "sha256-I1wwfn91hdH3jORIKny0Xc73qW2P04MjkVCgcaNnQUE=";
   };
```

Comment:
why are we vendoring `onnx`? it looks like there's an open MR to bump to 1.16 which would be compatible here https://github.com/NixOS/nixpkgs/pull/299313

Reply:
@acairncross, since you're listed as the `onnx` maintainer

Reply:
IIRC, onnxruntime is loosely correlated to onnx (the python package) so the don't have to be in lockstep. Although, I would probably prefer that we try to move to a world where we don't vendor a local copy.

Reply:
so i can open a PR updating `onnx` to 1.16.1 & then reference the source here; is there a preference for how to structure that?

i guess `python3Packages` is already an argument, so it could just be `python3Packages.onnx_1_16.src` & some comment or assertion indicating that we want compatibility checks in a similar fashion to how other packages with source dependencies like this are structured?

Reply:
There's a tradeoff, as now if you bump onnx, you have to be concerned about onnxruntime. And it's a matter of if that should be the case.